src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonObjectDefaultConverter.cs

```diff
@@ -11,7 +12,7 @@ namespace System.Text.Json.Serialization.Converters
     /// </summary>
     internal sealed class JsonObjectDefaultConverter<T> : JsonObjectConverter<T>
     {
-        internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, ref ReadStack state, out T value)
+        internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, ref ReadStack state, [MaybeNullWhen(false)] out T value)
```

Comment:
This may become protected (instead of just internal) and I don't think we want to say this shouldn't be null.

Also what's the difference between `DisallowNull` and `[MaybeNullWhen(false)]`?

Reply:
`DisallowNull` is for inputs, `MaybeNullWhen(false)` is for outputs. In this case this is stating that regardless of the nullability of the generic, the `OnTryRead` might set it to null when it returns false.

Even if we make it protected (and expose it), the nullability is showcasing the intent of the API. We are doing `T value = default!` in a bunch of places.

We can revisit the annotation if the implementation/design changes as part of exposing it. This change makes sense to me as it accurately represents what the API is doing so the caller knows what to do with the out parameter.